test/distributed/test_compute_comm_reordering.py

```diff
@@ -234,26 +233,24 @@ def func(a, *, tag, ranks, group_size):
             # 2. then, we schedule the ops (g) that ARE NOT required for second all_reduce and DO NOT depend on first all_reduce.
             # 3. then, we schedule the ops (f) that ARE required for second all_reduce and DO depend on first all_reduce.
             # and then, we schedule the second all_reduce. And then schedule all ops that depend on second all_reduce.
-            FileCheck().check("dist.all_reduce(").check("triton_poi_fused_relu").check(
-                "extern_kernels.mm("
-            ).check("extern_kernels.mm(").check("_wait_tensor(").check(
-                "triton_poi_fused_mul"
-            ).check(
-                "dist.all_reduce("
-            ).check(
-                "_wait_tensor("
-            ).check(
-                "triton_poi_fused_add"
-            ).check(
-                "extern_kernels.mm("
-            ).run(
-                code
+            (
```

Comment:
One thing I think we should add is - I think we should write a (small) simulator that simulates the expected overlap after our reordering/optimization passes. That would make these tests much cleaner to write.